src/main/java/spoon/support/reflect/declaration/CtClassImpl.java

```diff
@@ -69,8 +69,7 @@ public void accept(CtVisitor v) {
 		for (CtConstructor<T> c : constructors) {
 			boolean cont = c.getParameters().size() == parameterTypes.length;
 			for (int i = 0; cont && (i < c.getParameters().size()) && (i < parameterTypes.length); i++) {
-				if (!c.getParameters().get(i).getType().getQualifiedName()
-						.equals(parameterTypes[i].getQualifiedName())) {
+				if (!parameterTypes[i].equals(c.getParameters().get(i).getType())) {
```

Comment:
I read your PR and you fix 2 bugs in your PR, you should create another PR for this bug.